### ISSUE TITLE
Allow for syncing of contact ID and hash so mail recipients can update contact info in CiviCRM

### ISSUE THREAD

**Issue by herbdool** [veda-consulting/uk.co.vedaconsulting.mailchimp#159]
I think it would be a big benefit to also sync the contact ID and hash. Since it's not default in Mailchimp it would probably require manually creating it (unless the API can create it). This would open up the possibility of building links in Mailchimp so recipients can update their contact info in CiviCRM. 

I think that without this feature quite a few organizations are held back from using Mailchimp.

**Comment by veda-consulting** [comment]
@herbdool yes, we talked about this in the past. Its a feature we could build, we'd need funding for about 3 days development to make it happen.

**Comment by herbdool** [comment]
Nice! Would this also include the functionality to choose other fields to also sync with Mailchimp? I'm not sure how much harder that would be compared to the contact id/hash fields.

**Comment by herbdool** [comment]
I guess the tricky part is making sure the checksum is current. If it times out or gets replaced then the records in Mailchimp need to get updated.

**Comment by veda-consulting** [comment]
Would be great if you could create a wiki page about what you think needs to happen. Others can then also contribute to it and we can come up with a schedule of work and look for some funding.
What do you think?

**Comment by herbdool** [comment]
I think that's a good idea. Do you mean a wiki page in github?

**Comment by veda-consulting** [comment]
was thinking more on the civicrm wiki, not our git hub

**Comment by herbdool** [comment]
Okay, I took a look. I didn't see any obvious section to add a wiki page. Do you have an idea? I'll start one once I know that.

**Comment by sralsg** [comment]
Have there been any developments on the contact ID & hash sync issue? We'd definitely be interested in at least partly funding this.

**Comment by sunnycs121** [comment]
@veda-consulting Is this have been implemented or not yet? Can now custom fields be posted on mailchimp also now?

**Comment by veda-consulting** [comment]
Hi Hassan, not yet, it is on the roadmap but not done yet. We would welcome
any org who has some funds to contribute towards the work.
